blockmanager.go

```diff
@@ -2065,19 +2066,14 @@ func (b *blockManager) handleBlockchainNotification(notification *blockchain.Not
 
 	// A block has been disconnected from the main block chain.
 	case blockchain.NTBlockDisconnected:
```

Comment:
The handling in this path is certainly not correct since it's not using the state of the treasury agenda.  This also means there has not been any testing of reorgs done on simnet.

Reply:
Modified the code to receive the flag via function parameters and am now skipping the treasury base in the disconnect message.